### ISSUE TITLE
NIPS 2020 | Uncertainty-aware Self-training for Text Classification with Few Labels

### ISSUE THREAD

**Issue by richardbaihe** [richardbaihe/paperreading#65]
[This paper](https://papers.nips.cc/paper/2020/file/f23d125da1e29e34c552f448610ff25f-Paper.pdf) investigates the self-training method for the student-teacher data augmentation framework (label un-labeled data with teacher network, and then training student network with this pseudo-data).

The key problem is how to make the student network be aware of the pseudo-label's `uncertainty`, i.e. the student network should treat different pseudo-data differently in terms of their different uncertainty.

Hence, this paper proposes how to define uncertainty, how to sample data according to the uncertainty to train the student network, and how to train the uncertainty-aware student model (confident learning).
<img width="1398" alt="image" src="https://user-images.githubusercontent.com/11585856/110361765-74a80d00-800e-11eb-93e5-71a769c2dadd.png">

## 1. uncertainty
They use Bayesian Active Learning by Disagreement (BALD) to measure the entropy of teacher's ouputs and the agreement of outputs with different dropout parameters. The larger B is, the more uncertain the data is.
<img width="762" alt="image" src="https://user-images.githubusercontent.com/11585856/110361783-7c67b180-800e-11eb-8403-f4e10fec3002.png">

## 2. selection 
There are two strategies that can be used:
easier (select more data with lower uncertainty) and harder (select more data with higher uncertainty).
<img width="810" alt="image" src="https://user-images.githubusercontent.com/11585856/110361799-81c4fc00-800e-11eb-9fb1-8cfb5ffadd81.png">

## 3. confident learning

The above selection and uncertainty definition mainly focus on the mean of prediction. Here, the variance of predictions is incorporated into the final loss function. The following formula is wrong and should add a '-'.
<img width="774" alt="image" src="https://user-images.githubusercontent.com/11585856/110361827-8b4e6400-800e-11eb-9dc0-7e746341269b.png">
<img width="554" alt="image" src="https://user-images.githubusercontent.com/11585856/110361836-8ee1eb00-800e-11eb-967e-3499c0c7db67.png">

## 4. final results

<img width="853" alt="image" src="https://user-images.githubusercontent.com/11585856/110361856-95706280-800e-11eb-8711-21298333b495.png">

<img width="865" alt="image" src="https://user-images.githubusercontent.com/11585856/110361882-9acdad00-800e-11eb-864d-d34e4cea648b.png">
